### PR TITLE
PKE bootstrap commands

### LINKED ISSUE #1673: Create user data script calling banzai-kube install

### PR DESCRIPTION
**Pull request by orymate**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    |yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | closes #1673
| License         | Apache 2.0

### What's in this PR?
This PR adds the capability of creating Pipeline API tokens for the PKE installer (or other cluster service users), and to generate bootstrap commands that can be directly used as user-data scripts or from the CLI.

* New API endpoint: GET clusters/x/pke/commands
* Commands made available to EC2/PKE CreateCluster method for upcoming work

Limitation: jwt token gives general org-wide privileges instead of cluster-only

### Checklist

- [x] Implementation tested (with at least one cloud provider) -- PKE GKE
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [x] OpenAPI and Postman files updated, client regenerated (`make generate-client`) (if needed)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (needed!)

### To Do
- [x] test with a managed provider
- [x] add external url config to chart
- [x] `git rebase -i --autosquash origin/master`

### CODE REVIEW

File: cluster/common.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ type CommonCluster interface {
 	DeleteFromDatabase() error
 
 	// Cluster management
-	CreateCluster() error
+	CreateCluster(*Manager) error
```
Comment (orymate) [inline cluster/common.go]: no

File: api/cluster_create.go [hunk 2]
```diff
@@ -149,7 +149,7 @@ func (a *ClusterAPI) CreateCluster(
 		PostHooks:      postHooks,
 	}
 
-	creator := cluster.NewCommonClusterCreator(createClusterRequest, commonCluster)
+	creator := cluster.NewClusterCreator(createClusterRequest, commonCluster, a.tokenGenerator, a.externalBaseURL)
```
Comment (sagikazarmark) [inline api/cluster_create.go]: Let's create a separate function for the common and pke cluster creator and conditionally call them here.

Comment (orymate) [reply api/cluster_create.go]: why do you think that it's the api's responsibility to decide on how a specific type of clusters should be created?

Comment (sagikazarmark) [reply api/cluster_create.go]: It's not, but for now it's an acceptable compromise, until we figure out how to properly do this. Once we have a few more creators, having a single constructor will become unmaintainable. Refactoring a conditional logic is much easier.

**Pull request merged**